src/storage/patch/SparqlUpdatePatchHandler.ts

```diff
@@ -65,10 +65,25 @@ export class SparqlUpdatePatchHandler extends PatchHandler {
       throw new NotImplementedHttpError('WHERE statements are not supported');
     }
 
-    // Read the quads of the current representation
     const lock = await this.locker.acquire(identifier);
-    const quads = await this.source.getRepresentation(identifier,
-      { type: [{ value: INTERNAL_QUADS, weight: 1 }]});
+    try {
+      await this.applyPatch(identifier, deletes, inserts);
+    } finally {
+      await lock.release();
+    }
```

Comment:
That works indeed.
I wonder whether, for the future read/write interface, we want a `withReadLock(identifier, fn)` and `withWriteLock(identifier, fn)` structure to enforce this.

Any `acquire` without `try`/`finally` is essentially a mistake.